### PR TITLE
Add SSL Support to InfluxDB Connection

### PR DESCRIPTION
**Pull request by pradippatil**
This pull request adds support for secure SSL connections to InfluxDB. With SSL, data transmitted to InfluxDB will be encrypted, enhancing security and protecting metrics and data in transit.

This change will unblock users who need to connect to SSL-enabled InfluxDB instances, which currently fail with an error similar to the one mentioned below:

```bash
 InsecureRequestWarning: Unverified HTTPS request is being made to host. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
  warnings.warn(
---------------------------------------------------------------------------
InfluxDBClientError                       Traceback (most recent call last)

...
...

File [~/Documents/code/scripts/.venv/lib/python3.11/site-packages/influxdb/client.py:413]
in InfluxDBClient.write(self, data, params, expected_response_code, protocol)
    410         data = [data]
    411     data = ('\n'.join(data) + '\n').encode('utf-8')
--> 413 self.request(
    414     url="write",
    415     method='POST',
    416     params=params,
    417     data=data,
    418     expected_response_code=expected_response_code,
    419     headers=headers
    420 )
    421 return True

File [~/Documents/code/scripts/.venv/lib/python3.11/site-packages/influxdb/client.py:378]
in InfluxDBClient.request(self, url, method, params, data, stream, expected_response_code, headers)
    376 else:
    377     err_msg = reformat_error(response)
--> 378     raise InfluxDBClientError(err_msg, response.status_code)

InfluxDBClientError: 405: Method Not Allowed
```

### CODE REVIEW

File: src/grasshopper/lib/grasshopper.py [hunk 1]
```diff
@@ -67,8 +68,18 @@ def influx_configuration(self) -> dict[str, Optional[str]]:
         if pwd:
             configuration["pwd"] = pwd
 
+        configuration["ssl"] = self.to_bool(self.global_configuration.get("influx_ssl", "False"))
```
Comment (skezell) [inline src/grasshopper/lib/grasshopper.py:71]: I have some confusion here about why you are converting the value to bool here. The way that this variable is declared indicates that it should be automatically cast to a boolean (`"typecast": typecast_bool`). This occurs via the `typecast` fixture at a point after all values have been merged from all the different possible sources (env vars, command line, config files, etc.), thus ensuring that _all_ values will be the expected type, despite different incoming data types.

The `to_bool` portion is essentially a no op here. Either the value was supplied via the command line and converted as part of the` global_configuration` construction OR it's not in the collection and you can just as easily have the default value for your .get be False:

`configuration["ssl"] = self.global_configuration.get("influx_ssl", False)`

That being said, it is strongly preferred that the default value be handled in the attribute declaration, like so:
`        "influx_ssl": {
            "opts": ["--influx_ssl"],
            "attrs": {
                "action": "store",
                "type": bool,
                "help": "Enable SSL for InfluxDB connection.",
            },
            "typecast": typecast_bool,
            "default": False;
        }`

This style makes it easy to see everything about the config variables in one place, but also means that values can be correctly set via a variety of input sources (as noted above).

File: src/grasshopper/lib/grasshopper.py [hunk 2]
```diff
@@ -67,8 +68,18 @@ def influx_configuration(self) -> dict[str, Optional[str]]:
...
+        configuration["verify_ssl"] = self.to_bool(self.global_configuration.get("influx_verify_ssl", "False"))
+
         return configuration
 
+    @staticmethod
```
Comment (skezell) [inline src/grasshopper/lib/grasshopper.py:76]: As noted above, you don't really need this method. The `typecast_bool` method already exists and is automatically getting applied to all the variables.

File: README.md [hunk 3]
```diff
@@ -111,6 +111,9 @@ within that YAML file that corresponds to the scenario you wish to run. Defaults
 you must specify a host.
     E.g. `1.1.1.1`. Defaults to None.
 - `--influx_port`: Port for your `influx_host` in the case where it is non-default.
+- `--influx_ssl`: If your influxdb is using SSL, set this to True. Defaults to False.
+- `--influx_verify_ssl`: If your influxdb is using SSL, set this to True. Defaults to
```
Comment (skezell) [inline README.md:115]: Are these 2 different values or 2 different attributes? could one be False and one be True?
Could you be using ssl, but don't want the verify ssl to be true?

**Review by skezell — CHANGES_REQUESTED** [review]
Please move the default value to the attribute definition in gh_configuration.py.

File: README.md (hunk 3 above)
Comment (pradippatil) [reply README.md:115]: Yes, they are two different attributes. It is possible for one to be False and the other to be True. This allows for flexibility in situations where you want to use SSL but don’t require certificate verification

File: src/grasshopper/lib/grasshopper.py (hunk 1 above)
Comment (pradippatil) [reply src/grasshopper/lib/grasshopper.py]: I appreciate the clarification about the automatic typecasting to boolean via the typecast fixture. It seems I misunderstood the existing configuration setup. 

I've adjusted the code now.

File: README.md (hunk 3 above)
Comment (skezell) [reply README.md:115]: gotcha. thanks for clarifying.

**Review by skezell — APPROVED** [review]
Looks good!

**Review by jmfiola — APPROVED** [review]
LGTM, thanks for the contribution!!

**Comment by pradippatil** [comment]
Curious as to why no checks were run on this PR?

**Comment by jmfiola** [comment]
> Curious as to why no checks were run on this PR?

Hmm, great catch, not sure, but it does look like after merging, the ruff lint step failed. 

I will look into this today!

**Pull request merged**